flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskManagerServices.java

```diff
@@ -302,6 +299,30 @@ public static TaskManagerServices fromConfiguration(
 			taskEventDispatcher);
 	}
 
+	private static TaskSlotTable createTaskSlotTable(
+			int numberOfSlots,
+			long managedMemorySize,
+			long timerServiceShutdownTimeout) {
+		final List<ResourceProfile> resourceProfiles =
+			Collections.nCopies(numberOfSlots, computeSlotResourceProfile(numberOfSlots, managedMemorySize));
+		final TimerService<AllocationID> timerService = new TimerService<>(
+			new ScheduledThreadPoolExecutor(1),
+			timerServiceShutdownTimeout);
+		return new TaskSlotTable(createTaskSlotsFromResources(resourceProfiles), timerService);
+	}
+
+	public static List<TaskSlot> createTaskSlotsFromResources(Collection<ResourceProfile> resourceProfiles) {
```

Comment:
why is this public?

Reply:
used in `TaskSlotTableBuilder`, true, not needed